### PR TITLE
PAYARA-3368 Streamlined JACC module organization

### PR DESCRIPTION
**Pull request by arjantijms**
This is a refactoring and cleanup, no functional changes.

### CODE REVIEW

File: nucleus/core/logging:target:test-events.log [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[#|2018-12-16T13:54:32.990+0100|INFO||javax.enterprise.test.logging.events|_ThreadID=1;_ThreadName=main;_TimeMillis=1544964872990;_LevelValue=800;|Test message for testLogEventListenerNotifications|#]
```
Comment (Pandrex247) [inline nucleus/core/logging:target:test-events.log]: Err...?

File: appserver/security/jacc.provider.file/pom.xml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        </dependency>
+        
+        <!-- For the group to role mapper (SecurityRoleMapperFactory) -->
+        <dependency>
+            <groupId>org.glassfish.main.security</groupId>
+            <artifactId>security</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+	
+		<!-- The main entry point to the JACC provider is an HK2 service (PolicyConfigurationFactoryImpl)  -->
```
Comment (Pandrex247) [inline appserver/security/jacc.provider.file/pom.xml]: Please don't use tabs

File: nucleus/core/logging:target:test-events.log (hunk 1 above)
Comment (Cousjava) [inline nucleus/core/logging:target:test-events.log]: What's this doing here?

Comment (MattGill98) [inline nucleus/core/logging:target:test-events.log]: I think this file was accidentally committed!

File: appserver/security/jacc.provider.file/src/main/java/com/sun/enterprise/security/provider/PolicyUtil.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+import java.net.URL;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.util.Arrays;
+
+import sun.net.www.ParseUtil;
+import sun.security.util.Debug;
+import sun.security.util.Password;
```
Comment (Cousjava) [inline appserver/security/jacc.provider.file/src/main/java/com/sun/enterprise/security/provider/PolicyUtil.java]: It's generally a bad idea to use sun.* classes, is there an alternative?

File: appserver/security/jacc.provider.file/pom.xml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    its licensees as provided above.  However, if you add GPL Version 2 code
+    and therefore, elected the GPL Version 2 license, then the option applies
+    only if the new code is made subject to such option by the copyright
+    holder.
+
+-->
+
+<!-- Portions Copyright [2018] [Payara Foundation and/or its affiliates] -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (Cousjava) [inline appserver/security/jacc.provider.file/pom.xml]: New module in a 'refactor'?

File: appserver/distributions/pom.xml [hunk 5]
```diff
@@ -205,6 +208,34 @@
     </build>
     
     <profiles>
+
```
Comment (Cousjava) [inline appserver/distributions/pom.xml]: What do these changes have to do with JACC?

File: nucleus/core/logging:target:test-events.log (hunk 1 above)
Comment (arjantijms) [reply nucleus/core/logging:target:test-events.log]: eerrrhh.... where did THAT came from? :O

File: appserver/security/jacc.provider.file/pom.xml (hunk 2 above)
Comment (arjantijms) [reply appserver/security/jacc.provider.file/pom.xml]: Indeed... thx for spotting it!

File: appserver/security/jacc.provider.file/pom.xml (hunk 4 above)
Comment (arjantijms) [reply appserver/security/jacc.provider.file/pom.xml]: @Cousjava 

Yes, it was moved (factored out) from the main ee security module. See issue description.

File: appserver/distributions/pom.xml (hunk 5 above)
Comment (arjantijms) [reply appserver/distributions/pom.xml]: I added the factored out module to several of those files, and to make the naming consistent and easy to spot as being JACC the inmemory module was renamed to start with JACC, which needed to be updated in these files. The security modules were grouped a bit more tighter together as well. Comments were added were necessary (see original issue), and the odd ordering or inconsistent comment was addressed when looking at these files as well.

File: appserver/security/jacc.provider.file/src/main/java/com/sun/enterprise/security/provider/PolicyUtil.java (hunk 3 above)
Comment (arjantijms) [reply appserver/security/jacc.provider.file/src/main/java/com/sun/enterprise/security/provider/PolicyUtil.java]: @Cousjava Part of the JDK 11 effort, but the mini-step is to isolate those dependencies from the main EE module to this plug-in module.

**Comment by arjantijms** [comment]
Jenkins test please

**Pull request merged**